### PR TITLE
Feat: Add optional scale bar to `viewer` with mathematical safety checks

### PR DESCRIPTION
**Pull request by marcosdanieldasilva**
**Description**
This PR introduces an optional `scale` keyword argument (default `false`) to the `viewer` function in `GeoTablesMakieExt`. When enabled, it dynamically renders a linear scale bar directly on the plot, providing essential spatial context for fields like forestry, geology, and agriculture.

**Implementation Details**
- **Internal Helper:** Implemented `addscalebar!` as an internal function within the extension to avoid namespace pollution and keep the implementation lightweight.
- **Mathematical Safety (Validation):** A linear 2D scale bar is mathematically invalid for 3D perspective views and unprojected angular coordinates (e.g., degrees). The implementation uses a strict `embeddim(dom) === 3` check. Because the `Meshes.jl` ecosystem inherently embeds geographic surfaces (`LatLon`) into a 3D spatial datum, this single condition elegantly and natively intercepts both true 3D volumetric grids and unprojected geographic maps.
- **Informative Warnings:** If `scale=true` is requested on an invalid domain, the plot still renders successfully, but the scale bar is safely skipped and a clear `@warn` is issued explaining the cartographic/dimensional limitation.

**Test Scenarios Evaluated Locally**
- [x] **2D Cartesian / Implicit Grids:** Successfully draws the scale bar.
- [x] **2D Projected CRS (e.g., WebMercator, UTM):** Successfully draws the scale bar.
- [x] **2D Geographic CRS (e.g., LatLon, WGS84):** Safely caught by `embeddim === 3`. Scale bar skipped, warning issued.
- [x] **3D Volumetric Grids:** Safely caught by `embeddim === 3`. Scale bar skipped, warning issued.
- [x] **Default state (`scale=false`):** Standard plotting remains unaffected with zero performance overhead.

**Pending Items**
- The formal test suite for these changes will be added in subsequent commits.
- 
> Note: The manual test script used to generate and validate these visual edge cases locally is available here: https://github.com/marcosdanieldasilva/GeoTests/blob/main/src/testsforscalebar.jl

### CODE REVIEW

File: ext/GeoTablesMakieExt/GeoTablesMakieExt.jl [hunk 1]
```diff
@@ -14,7 +14,7 @@ using DataScienceTraits
 using CategoricalArrays
 using Distributions: Distribution, location
 using Makie.Colors: Colorant, Gray
-using Meshes: embeddim
+using Meshes: embeddim, crs
```
Comment (juliohm) [inline ext/GeoTablesMakieExt/GeoTablesMakieExt.jl]: It seems `crs` is not used in the proposed changes?

File: ext/GeoTablesMakieExt/viewer.jl [hunk 2]
```diff
@@ -2,7 +2,62 @@
 # Licensed under the MIT License. See LICENCE in the project root.
 # ------------------------------------------------------------------
 
-function viewer(data::AbstractGeoTable; alpha=nothing, colormap=nothing, colorrange=nothing, kwargs...)
+function addscalebar!(axis; position=(0.85, 0.05), targetaxfrac=0.25, color=:black, linewidth=3.0, fontsize=16)
```
Comment (juliohm) [inline ext/GeoTablesMakieExt/viewer.jl]: Please move this auxiliary function to after `viewer`, and rename it to `scalebar!`.

File: ext/GeoTablesMakieExt/viewer.jl [hunk 3]
```diff
@@ -2,7 +2,62 @@
...
+    lengthdata = mul
+
+    # relative length (0-1)
+    lengthrel = lengthdata / safewidth
+
+    avgpos = Makie.Point2f(position)
+    p1 = avgpos - Makie.Vec2f(lengthrel / 2, 0)
+    p2 = avgpos + Makie.Vec2f(lengthrel / 2, 0)
+
+    return (points=[p1, p2], text="$(mul)", textpos=avgpos)
```
Comment (juliohm) [inline ext/GeoTablesMakieExt/viewer.jl]: We don't use explicit `return`

File: ext/GeoTablesMakieExt/viewer.jl [hunk 4]
```diff
@@ -2,7 +2,62 @@
 # Licensed under the MIT License. See LICENCE in the project root.
 # ------------------------------------------------------------------
 
-function viewer(data::AbstractGeoTable; alpha=nothing, colormap=nothing, colorrange=nothing, kwargs...)
+function addscalebar!(axis; position=(0.85, 0.05), targetaxfrac=0.25, color=:black, linewidth=3.0, fontsize=16)
+  muls = [
+    p isa Int ? x * p : round(x * p, sigdigits=4) for
+    p in Real[[10.0^p for p in -50:-1]; [1, 10, 100, 1000, 10000]; [10.0^p for p in 5:50]] for x in [1, 2, 5]
+  ]
```
Comment (juliohm) [inline ext/GeoTablesMakieExt/viewer.jl]: Could you please comment on this `muls` variable?

File: ext/GeoTablesMakieExt/viewer.jl [hunk 5]
```diff
@@ -2,7 +2,62 @@
...
+    yautolimits=false
+  )
+end
+
+function viewer(
+  data::AbstractGeoTable;
+  alpha=nothing,
+  colormap=nothing,
+  colorrange=nothing,
+  scale::Bool=false,
```
Comment (juliohm) [inline ext/GeoTablesMakieExt/viewer.jl]:
```suggestion
  scalebar=nothing,
```

Comment (juliohm) [reply ext/GeoTablesMakieExt/viewer.jl]: We should set `scalebar` to `nothing` by default in order to decide the default based on the input data. If the user is explicit about `scalebar`, either `true` or `false`, then we skip the code that computes the default:

```julia
sbar = isnothing(scalebar) ? defaultscalebar(data) : scalebar
```

**Review by juliohm — CHANGES_REQUESTED** [review]
Thank you @marcosdanieldasilva ! Attached a first round of review.

File: ext/GeoTablesMakieExt/viewer.jl (hunk 5 above)
Comment (marcosdanieldasilva) [reply ext/GeoTablesMakieExt/viewer.jl]: I didn’t fully understand, could you check if the changes I just made match what was requested?

**Comment by juliohm** [comment]
Thank you @marcosdanieldasilva. I've tested it locally and it is nice to see it working already. I have a couple of suggestions to improve the feature:

1. I see the function takes an `axis` as input. I wonder if we should try to pass a `fig` like we do with the other `cbar` utility, for consistency. If that is the case, we could rename the function to `sbar` and export it as an additional function that users could use in their own visualizations outside the `viewer`.
2. If possible, could we place the scale bar outside the axis? Perhaps by creating a new `Makie.Axis` as a subfigure with shared x axis (so that they track the zoom)? That would facilitate reading the scale bar without interference from objects already in the scene.
3. We need to add the units of measurement in the scale bar. You can retrieve the unit from the domain with `unit(lentype(dom))`. That object should have a string representation already so you should just concatenate the string in the `Makie.text!` call.

**Comment by juliohm** [comment]
@marcosdanieldasilva I will close this PR for now because I am doing a major refactor in our viewer to increase support for exotic objects. We can restart this work in a fresh PR in the future.
